### PR TITLE
Farming: Make cotton look like cotton, add crafted string item

### PR DESCRIPTION
**Pull request by paramat**
Addresses our oldest issue #404 

Change farming:cotton item texture to look like cotton not string, texture by Napiphelios.
Add farming:string item crafted from farming:cotton.
Remove old alias `minetest.register_alias("farming:string", "farming:cotton")` it has been active for years and items tend to be converted faster than nodes. If anyone still has a stash of farming:string anywhere it is now supported as an item again.

Possible breakage:
Mods that use farming:cotton as a string item should now use farming:string, however the breakage is not severe as string is crafted from cotton 1 to 1 and is the same material, just the texture looks like a cotton ball now.
Any other possible breakage?
0.5 is a good chance to clean this mess up.

### CODE REVIEW

File: mods/farming/init.lua [hunk 1]
```diff
@@ -66,7 +78,16 @@ minetest.register_craft({
 	}
 })
 
+minetest.register_craft({
+	output = "farming:string 3",
+	recipe = {
+		{"farming:cotton", "farming:cotton", "farming:cotton"},
+	}
+})
```
Comment (DS-Minetest) [inline mods/farming/init.lua]: Imho a vertical row would be better. 3 items of the same type in a horizontal row reminds to a slab recipe. But eg. technic also has a recipe to make a wire item vertical.

**Comment by cx384** [comment]
I think you should need a spinning wheel to make string.

File: mods/farming/init.lua (hunk 1 above)
Comment (paramat) [inline mods/farming/init.lua]: Hmm maybe.

**Comment by paramat** [comment]
Tested.

**Comment by DS-Minetest** [comment]
@cx384 I think that would be too complex for MTG and there will be the same problems as for the renaming workbench.

**Comment by TumeniNodes** [comment]
Spinning wheel would be a bit much (at least currently)
The current set up is clean and simple, and makes sense... and works.
I don't think it really matters which orientation the crafting calls for since it's not likely going to result in a string_slab .

**Comment by cx384** [comment]
@DS-Minetest yeah, but a spinning wheel would fit in the medieval concept of minetest and it would enhance the Immersion.
I know I spinning wheel ... well, a wheel is circular ... but, imagine the spinning wheel as a craftitem which you need to craft string.

**Comment by TumeniNodes** [comment]
that could be done from another mod, which could also add even more uses and crafts for string

**Comment by cx384** [comment]
OK but without mods string doesn't have any use. Therefore there is no significant reason to ad string to the minetest_game.

**Comment by TumeniNodes** [comment]
This could be considered the "first step" then.
This will get a very old issue out of the way, with a decent solution.
And "then" maybe ideas could come after for a use for string. Maybe such as adding bows to tools (for hunting) adding one bow to each tool materials?
But, as stated, getting this simple matter settled first is a good step.

**Comment by paramat** [comment]
Currently cotton is already string, so we're not adding string, many mods use the string item from MTG.

**Comment by TumeniNodes** [comment]
I think @cx384 is reference to having uses for string within MTG, itself?
But that is a minor thing which could be discussed another time.
I feel this sort of fix should be merged a.s.a.p. to knock it out of the way. Tested and works fine.

IMO 2 MTG devs should be able to take something this trivial (but needed) and merge it without a week long (or longer discussion session) this is the sort of thing which causes drag and stalemate.
Gets put on hold or back-burner then they add up quick

**Comment by paramat** [comment]
Looking back, farming mod used to have cotton plants drop a farming:string item (with a string texture). More recently they drop farming:cotton (with a string texture) and there's an alias from farming:string to farming:cotton. So i don't see any serious breakage.

**Comment by TumeniNodes** [comment]
there is no breakage

**Comment by paramat** [comment]
Thanks will use.

**Comment by paramat** [comment]
LOL it's down from 318 to 316 bytes, but will still use it.

**Comment by TumeniNodes** [comment]
haha, I know but ... "optimized"
I'm going to start going through all the textures soon anyway, so this will be one more down

**Comment by paramat** [comment]
So optimisation has benefit even if there's no file size decrease? Maybe more efficient decompression?

File: mods/farming/init.lua (hunk 1 above)
Comment (paramat) [inline mods/farming/init.lua]: Maybe 2 in a row so that 3 do not have to be crafted each time?

**Comment by paramat** [comment]
Added your texture.

**Comment by TumeniNodes** [comment]
It's still Napiophelios' texture, just 2bytes lighter in file size.
And tbh, that 2bytes does not make a difference and was done solely as this seems to be desired.
I figured I would nab this image while it was fresh in the pond but, tbh again, should be very careful with optimizing, it could end up doing more harm than good?

better a discussion elsewhere regarding that though at some point. there is a lot to the idea of using optimization (pros and cons) so it definitely needs more thinking/figuring, and I am definitely not a pro

**Comment by paramat** [comment]
I assume optipng. Optimisation does not, as far as i know, alter the visible pixels in any way, so i can't see any harm.
The only issue is with 'allfaces optional' drawtype nodes, such as leaves, that need to retain a colour for the transapent pixels for 'opaque' mode. We have found those need to stay RGBA instead of indexed, and any opitmisation should be tested in 'opaque' leaf mode.

**Comment by TumeniNodes** [comment]
yes, optipng. pngcrush seems overboard for what is needed here.
I will comment on a few things in Issues #1831 seems a more appropriate place for this topic

btw, this PR should already have approvals and merged (just sayin) tests fine, looks good. heh

**Comment by paramat** [comment]
@sofar

**Comment by paramat** [comment]
Updated, 2 string returned from cotton above cotton, vertical used instead of horizontal.

**Comment by paramat** [comment]
Approval from SmallJoker (Krock) http://irc.minetest.net/minetest-dev/2017-08-18#i_5047098

**Comment by paramat** [comment]
08727bcd698d511c880bd1c2f7b454bfe0c2b20d

**Pull request closed without merging**